crates/rpc/rpc-eth-types/src/revm_utils.rs

```diff
@@ -136,12 +134,17 @@ impl CallFees {
                             RpcInvalidTransactionError::TipAboveFeeCap.into(),
                         )
                     }
-                    Ok(min(
-                        max_fee,
+
+                    let tip = max_fee.saturating_sub(block_base_fee);
+                    let effective_price = if tip > max_priority_fee_per_gas {
                         block_base_fee.checked_add(max_priority_fee_per_gas).ok_or_else(|| {
                             EthApiError::from(RpcInvalidTransactionError::TipVeryHigh)
```

Comment:
`RpcInvalidTransactionError::TipVeryHigh` error seems wrong here

Reply:
this error would be thrown if the this overflows because the tip is too high so this seems appropriate